samples/client/petstore/java/retrofit2/src/main/java/io/swagger/client/JSON.java

```diff
@@ -39,8 +46,25 @@
     private OffsetDateTimeTypeAdapter offsetDateTimeTypeAdapter = new OffsetDateTimeTypeAdapter();
     private LocalDateTypeAdapter localDateTypeAdapter = new LocalDateTypeAdapter();
 
+    public static GsonBuilder createGson() {
+        GsonFireBuilder fireBuilder = new GsonFireBuilder()
+          .registerTypeSelector(Animal.class, new TypeSelector() {
+            @Override
+            public Class getClassForElement(JsonElement readElement) {
+                Map classByDiscriminatorValue = new HashMap();
+                classByDiscriminatorValue.put("Cat".toUpperCase(), Cat.class);
+                classByDiscriminatorValue.put("Dog".toUpperCase(), Dog.class);
```

Comment:
Isn't `Animal` a valid discriminator value ?

Reply:
I assumed my case with abstract superclass would be most typical. I added support for it and sample is including also `Animal`.